Please add supported Kernel 5.1

make clean
make:

make[1]: вход в каталог «/usr/src/linux-headers-5.1.0-050100-generic»
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_cmd.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_security.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_debug.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_io.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_ioctl_query.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_ioctl_set.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_ieee80211.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_mlme.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_mlme_ext.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_mi.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_wlan_util.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_vht.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_pwrctrl.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_rf.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_recv.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_sta_mgt.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_ap.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/mesh/rtw_mesh.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/mesh/rtw_mesh_pathtbl.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/mesh/rtw_mesh_hwmp.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_xmit.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_p2p.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_rson.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_tdls.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_br_ext.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_iol.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_sreset.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_btcoex_wifionly.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_btcoex.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_beamforming.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_odm.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_rm.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/rtw_rm_fsm.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/core/efuse/rtw_efuse.o
  CC [M]  /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.o
/home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.c: In function ‘isFileReadable’:
/home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.c:2190:10: error: implicit declaration of function ‘get_ds’; did you mean ‘get_da’? [-Werror=implicit-function-declaration]
   set_fs(get_ds());
          ^~~~~~
          get_da
/home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.c:2190:10: error: incompatible type for argument 1 of ‘set_fs’
In file included from ./include/linux/uaccess.h:11:0,
                 from ./include/linux/crypto.h:26,
                 from ./include/crypto/hash.h:16,
                 from ./include/linux/uio.h:14,
                 from ./include/linux/socket.h:8,
                 from ./include/linux/compat.h:15,
                 from ./include/linux/ethtool.h:17,
                 from ./include/linux/netdevice.h:41,
                 from /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/include/osdep_service_linux.h:30,
                 from /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/include/osdep_service.h:50,
                 from /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/include/drv_types.h:27,
                 from /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.c:19:
./arch/x86/include/asm/uaccess.h:29:20: note: expected ‘mm_segment_t {aka struct <anonymous>}’ but argument is of type ‘int’
 static inline void set_fs(mm_segment_t fs)
                    ^~~~~~
/home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.c: In function ‘retriveFromFile’:
/home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.c:2228:11: error: incompatible type for argument 1 of ‘set_fs’
    set_fs(get_ds());
           ^~~~~~
In file included from ./include/linux/uaccess.h:11:0,
                 from ./include/linux/crypto.h:26,
                 from ./include/crypto/hash.h:16,
                 from ./include/linux/uio.h:14,
                 from ./include/linux/socket.h:8,
                 from ./include/linux/compat.h:15,
                 from ./include/linux/ethtool.h:17,
                 from ./include/linux/netdevice.h:41,
                 from /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/include/osdep_service_linux.h:30,
                 from /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/include/osdep_service.h:50,
                 from /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/include/drv_types.h:27,
                 from /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.c:19:
./arch/x86/include/asm/uaccess.h:29:20: note: expected ‘mm_segment_t {aka struct <anonymous>}’ but argument is of type ‘int’
 static inline void set_fs(mm_segment_t fs)
                    ^~~~~~
/home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.c: In function ‘storeToFile’:
/home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.c:2263:11: error: incompatible type for argument 1 of ‘set_fs’
    set_fs(get_ds());
           ^~~~~~
In file included from ./include/linux/uaccess.h:11:0,
                 from ./include/linux/crypto.h:26,
                 from ./include/crypto/hash.h:16,
                 from ./include/linux/uio.h:14,
                 from ./include/linux/socket.h:8,
                 from ./include/linux/compat.h:15,
                 from ./include/linux/ethtool.h:17,
                 from ./include/linux/netdevice.h:41,
                 from /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/include/osdep_service_linux.h:30,
                 from /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/include/osdep_service.h:50,
                 from /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/include/drv_types.h:27,
                 from /home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.c:19:
./arch/x86/include/asm/uaccess.h:29:20: note: expected ‘mm_segment_t {aka struct <anonymous>}’ but argument is of type ‘int’
 static inline void set_fs(mm_segment_t fs)
                    ^~~~~~
cc1: some warnings being treated as errors
scripts/Makefile.build:275: recipe for target '/home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.o' failed
make[2]: *** [/home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959/os_dep/osdep_service.o] Error 1
Makefile:1571: recipe for target '_module_/home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959' failed
make[1]: *** [_module_/home/vav4eg/wifi/rtl88x2BU_WiFi_linux_v5.3.1_27678.20180430_COEX20180427-5959] Error 2
make[1]: выход из каталога «/usr/src/linux-headers-5.1.0-050100-generic»
Makefile:1996: recipe for target 'modules' failed
make: *** [modules] Error 2

--- a/os_dep/osdep_service.c
+++ b/os_dep/osdep_service.c
@@ -2187,7 +2187,7 @@ static int isFileReadable(const char *path, u32 *sz)
 		ret = PTR_ERR(fp);
 	else {
 		oldfs = get_fs();
-		set_fs(get_ds());
+		set_fs(KERNEL_DS);
 
 		if (1 != readFile(fp, &buf, 1))
 			ret = PTR_ERR(fp);
@@ -2225,7 +2225,7 @@ static int retriveFromFile(const char *path, u8 *buf, u32 sz)
 			RTW_INFO("%s openFile path:%s fp=%p\n", __FUNCTION__, path , fp);
 
 			oldfs = get_fs();
-			set_fs(get_ds());
+			set_fs(KERNEL_DS);
 			ret = readFile(fp, buf, sz);
 			set_fs(oldfs);
 			closeFile(fp);
@@ -2260,7 +2260,7 @@ static int storeToFile(const char *path, u8 *buf, u32 sz)
 			RTW_INFO("%s openFile path:%s fp=%p\n", __FUNCTION__, path , fp);
 
 			oldfs = get_fs();
-			set_fs(get_ds());
+			set_fs(KERNEL_DS);
 			ret = writeFile(fp, buf, sz);
 			set_fs(oldfs);
 			closeFile(fp);

--- a/os_dep/osdep_service.c
+++ b/os_dep/osdep_service.c
@@ -2187,7 +2187,11 @@ static int isFileReadable(const char *path, u32 *sz)
 		ret = PTR_ERR(fp);
 	else {
 		oldfs = get_fs();
+		#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0))
 		set_fs(KERNEL_DS);
+		#else
+		set_fs(get_ds());
+		#endif
 
 		if (1 != readFile(fp, &buf, 1))
 			ret = PTR_ERR(fp);
@@ -2225,7 +2229,11 @@ static int retriveFromFile(const char *path, u8 *buf, u32 sz)
 			RTW_INFO("%s openFile path:%s fp=%p\n", __FUNCTION__, path , fp);
 
 			oldfs = get_fs();
+			#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0))
 			set_fs(KERNEL_DS);
+			#else
+			set_fs(get_ds());
+			#endif
 			ret = readFile(fp, buf, sz);
 			set_fs(oldfs);
 			closeFile(fp);
@@ -2260,7 +2268,11 @@ static int storeToFile(const char *path, u8 *buf, u32 sz)
 			RTW_INFO("%s openFile path:%s fp=%p\n", __FUNCTION__, path , fp);
 
 			oldfs = get_fs();
+			#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0))
 			set_fs(KERNEL_DS);
+			#else
+			set_fs(get_ds());
+			#endif
 			ret = writeFile(fp, buf, sz);
 			set_fs(oldfs);
 			closeFile(fp);

